File: backend/src/main/java/com/festago/dto/StageResponse.java
package com.festago.dto;

import com.festago.domain.Stage;
import java.time.LocalDateTime;

public record StageResponse(Long id, LocalDateTime startTime) {

    public static StageResponse from(Stage stage) {
        return new StageResponse(stage.getId(), stage.getStartTime());
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/StageResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.model.Stage
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class StageResponse(
    val id: Int,
    val startTime: String,
) {
    fun toDomain(): Stage = Stage(
        id = id,
        startTime = LocalDateTime.parse(startTime),
    )
}
